### ISSUE TITLE
no-magic-numbers ignore calculations

### ISSUE THREAD

**Issue by AdriVanHoudt** [eslint/eslint#4226]
I have a const like `const oneDay = 1000 * 60 * 60 * 24;`
With `"no-magic-numbers": [2, { ignore: [1000] }]"
`No magic number: 60 [no-magic-numbers]	const oneDay = 1000 * 60 * 60 * 24;`

The point of doing the `oneDay` was to not have a magic number/calculation, would it be possible to add an option to ignore magic numbers that are directly behind an assignment?

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by AdriVanHoudt** [comment]
Eslint v1.7.3 btw

**Comment by AdriVanHoudt** [comment]
Sure but to me the oneDay is desciptive enough

**Comment by lo1tuma** [comment]
No, `oneDay` doesn’t explain the meaning of e.g. `60`. So I don’t think an exception for `no-magic-numbers` makes sense, because detecting such numbers is exactly the purpose of the rule.

**Comment by AdriVanHoudt** [comment]
Yeah but I was proposing an option so that these multiplications  would be considered 1 number

**Comment by nzakas** [comment]
I'd suggest disabling the rule for that line.

**Comment by AdriVanHoudt** [comment]
The thing is it occurs on more than 1 line and I do not want to bloat my code all over the place with temp disabling this rule

**Comment by gyandeeps** [comment]
Why not just create a function which can be consumed from different spots:

```js
function getMillisecondsPerDay(){
    const millisecondsPerSecond = 1000;
    const secondsPerMinute = 60;
    const minutesPerHour = 60;
    const hoursPerDay = 24;
    const millisecondsPerDay = millisecondsPerSecond * secondsPerMinute * minutesPerHour * hoursPerDay;

    return millisecondsPerDay;
}
```

**Comment by IanVS** [comment]
the number of milliseconds in a day is a constant, so just create it as a constant.  Why do you need to make the computer do your math for you?  You can explain with a comment if you really feel the need.

```js
const millisecondsPerDay = 86400000; // 24 * 60 * 60 * 1000
```

**Comment by AdriVanHoudt** [comment]
@ianvs why have I not thought of that :D

**Comment by keithamus** [comment]
I would request this rule for instances where a simple calculation is needed. Take for example the following code:

```js
const myArray = [1,2,3];
const lastIndex = myArray.length - 1;
```

This const uses 1 number as part of the calculation, and should be allowed to be passed through - because the number on its own has no semantic value, and trying to give it a useful name becomes a struggle, e.g:

```js
const myArray = [1,2,3];
const lastIndexLengthDelta = 1;
const lastIndex = myArray.length - lastIndexLengthDelta;
```

I'd argue the first example, while not attaching a semantic value to the number, is more readable than the 2nd. Thoughts?
